content/silver/ref/expr/pattern-matching.md

```diff
@@ -121,9 +143,35 @@ end
 ```
 The relative order of patterns for forwarding and non-forwarding productions is relevant for which pattern matches.
 In this example, the `d(_)` pattern will never be reached because any trees built by the `d` production will match the `a(_)` pattern.
+This is the same as placing a more general pattern before a more specific pattern in non-forwarding pattern matching.
 To match a forwarding production, place it at the beginning of the clauses.
 It is not an error to place it later, but it is unlikely the resulting semantics are the intended ones.
 
+Matching through forwarding occurs on a production basis, not a pattern basis.
+To see what we mean by this, suppose we have the following production:
+```
+abstract production foo
+top::Nt ::= c::Integer
+{
+  forwards to if c == 1 then foo(0) else bar(c);
+}
+```
+We try to match against the `foo` production with this `case` expression:
+```
+case x of
+| foo(0) -> -1
+| foo(a) -> a
+| _ -> 0
+end
+```
+If `x = foo(1)`, we match the second rule, `foo(a) -> a`.
```

Comment:
I'd discuss the attempt to match patterns in the order in which they appear.  First `foo(0)` fails ... then we try `foo(a)`...